src/main/java/org/testng/DependencyMap.java

```diff
@@ -21,7 +22,20 @@
 
   public DependencyMap(ITestNGMethod[] methods) {
     for (ITestNGMethod m : methods) {
-    	m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
+      m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
```

Comment:
The method from the parent class is excluded as it is overridden, so only this method from the child should be present in `methods` argument. If I make this method from the parent class be present in `methods` argument then this method is present in the executed tests list, so there are 3 tests executed instead of 2, which is wrong